src/game_api/layer.hpp

```diff
@@ -1,13 +1,19 @@
 #pragma once
-#include "entity.hpp"
 
-struct Entities
+class Entity;
+
+struct EntityList
 {
     Entity** ent_list;
     uint32_t* uids;
     uint32_t cap;
     uint32_t size;
 
+    constexpr bool empty()
+    {
+        return !!size;

```

Comment:
Please don't start using `!!` but be explicit `size == 0`

Reply:
This was just copied straight from the `Vector` type

Reply:
Then it's a good opportunity to make it nicer  :)